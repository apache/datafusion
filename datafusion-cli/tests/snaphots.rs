use insta::glob;
use insta_cmd::{assert_cmd_snapshot, get_cargo_bin};
use std::process::Command;
use std::{env, fs};

fn cli() -> Command {
    Command::new(get_cargo_bin("datafusion-cli"))
}

#[test]
fn test_snapshots() {
    if env::var("TEST_INTEGRATION").is_err() {
        eprintln!("Skipping integration tests");
        return;
    }

    let mut settings = insta::Settings::clone_current();
    settings.add_filter(r"Elapsed .* seconds\.", "[ELAPSED]");
    settings.add_filter(r"DataFusion CLI v.*", "[CLI_VERSION]");

    let _bound = settings.bind_to_scope();

    glob!("sql/*.sql", |path| {
        let input = fs::read_to_string(path).unwrap();
        let file = path.file_stem().unwrap().to_string_lossy();

        assert_cmd_snapshot!(file.as_ref(), cli().pass_stdin(input))
    });
}
