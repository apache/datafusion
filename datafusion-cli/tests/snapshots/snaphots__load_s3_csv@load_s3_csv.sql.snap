---
source: tests/snaphots.rs
info:
  program: datafusion-cli
  args: []
  stdin: "CREATE EXTERNAL TABLE cars\nSTORED AS CSV\nOPTIONS(\n    'aws.access_key_id' 'test',\n    'aws.secret_access_key' 'test',\n    'aws.region' 'us-east-1',\n    'aws.endpoint' 'http://localhost:4566',\n    'aws.allow_http' 'true'\n)\nLOCATION 's3://test-bucket/cars.csv';\n\nselect * from cars limit 1;"
input_file: tests/sql/load_s3_csv.sql
---
success: true
exit_code: 0
----- stdout -----
[CLI_VERSION]
0 row(s) fetched. 
[ELAPSED]

+-----+-------+---------------------+
| car | speed | time                |
+-----+-------+---------------------+
| red | 20.0  | 1996-04-12T12:05:03 |
+-----+-------+---------------------+
1 row(s) fetched. 
[ELAPSED]

\q

----- stderr -----
